pennylane/estimator/templates/subroutines.py

```diff
@@ -971,6 +971,174 @@ def resource_decomp(cls, base_cmpr_op, num_iter):
         return gate_counts
 
 
+class UnaryIterationBasedQPE(ResourceOperator):
+    r"""Resource class for UnaryIterationBasedQPE.
+
+    Args:
+        unitary (:class:`~.pennylane.estimator.resource_operator.ResourceOperator`): the unitary operator to run
+            the phase estimation protocol on
+        num_iterations (int): the number of iterations
+        adj_qft_op (:class:`~.pennylane.estimator.resource_operator.ResourceOperator` | None): An optional
+            argument to set the subroutine used to perform the adjoint QFT operation.
+        wires (Sequence[int], None): the wires the operation acts on
+
+    Resources:
```

Comment:
Is there any more we can or should say about the resources?
* Does this technique use the unary iterator from https://arxiv.org/pdf/1805.03662 just like [MultiControlledX](https://docs.pennylane.ai/en/stable/code/api/pennylane.estimator.ops.MultiControlledX.html)?
* Does it improve over any of the other QPEs we have in some specific way?
* Are there some advantages/disadvantages and tradeoffs?